doc/packages/build-support.md

```diff
@@ -0,0 +1,99 @@
+# Build Support {#sec-build-support}
+
+## `pkgs.substitute` {#pkgs-substitute}
+
+`pkgs.substitute` is a wrapper around [the `substitute` function](#fun-substitute) provided by the
+stdenv. It replaces strings in `src` as specified by the `substitutions` argument.
+
+In a build script, the line:
+
+```bash
+substitute $infile $outfile --replace-fail @foo@ ${foopkg}/bin/foo
+```
+
+is equivalent to:
+
+```nix
+{ substitute, foopkg }:
+substitute {
+  src = ./sourcefile.txt;
+  substitutions = [
+    "--replace"
+    "@foo@"
+    "${foopkg}/bin/foo"
+  ];
+}
+```
+
+## `pkgs.substituteAll` {#pkgs-substituteall}
+
+`pkgs.substituteAll` substitutes all instances of `@varName@` (`@`s included)
+in file file `src` with the value of the corresponding environment variable. As
```

Comment:
Somehow duplicate, maybe because i did a wrong suggestion.

```suggestion
in file `src` with the value of the corresponding environment variable. As
```

Reply:
Still there